agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -195,11 +196,59 @@ public String toString() {
     public LiveMetrics liveMetrics = new LiveMetrics();
     public LegacyRequestIdPropagation legacyRequestIdPropagation = new LegacyRequestIdPropagation();
 
+    public List<InheritedAttribute> inheritedAttributes = new ArrayList<>();
+
     public ProfilerConfiguration profiler = new ProfilerConfiguration();
     public GcEventConfiguration gcEvents = new GcEventConfiguration();
     public AadAuthentication authentication = new AadAuthentication();
   }
 
+  public static class InheritedAttribute {
+    public String key;
+    public SpanAttributeType type;
+
+    public AttributeKey<?> getAttributeKey() {
+      switch (type) {
+        case STRING:
+          return AttributeKey.stringKey(key);
+        case BOOLEAN:
+          return AttributeKey.booleanKey(key);
```

Comment:
i checked out #1743, the way they send extra attributes are through part C.  Breeze accepts key/value pair string only.  how other datatypes here will get used?

Reply:
they are all converted to strings, see `Exporter.getStringValue()`